tf2/include/tf2/buffer_core.h

```diff
@@ -382,12 +382,15 @@ class BufferCore : public BufferCoreInterface
   std::string allFramesAsStringNoLock() const;
 
   bool setTransformImpl(
-    const tf2::Transform & transform_in, const std::string frame_id,
-    const std::string child_frame_id, const TimePoint stamp,
+    const tf2::Vector3 & origin_in, const tf2::Quaternion & rotation_in, const std::string & frame_id,
```

Comment:
note: I changed the strings into references - this really should be a `std::string_view` - it _shouldn't_ break any code, and ROS is c++17 now - but it's somewhat a moot point as SSO will kick in for most `char*` conversions.